pkg/smokescreen/smokescreen.go

```diff
@@ -421,15 +421,74 @@ func hasPort(s string) bool {
 	return strings.LastIndex(s, "]") < strings.LastIndex(s, ":")
 }
 
-// normalizeHost returns host (as string) and port (as int) derived from the
+func _normalizePort(s string) (port int, err error) {
+	if strings.HasPrefix(s, "0") && s != "0" {
```

Comment:
Is the conversion using `strconv.Atoi()` not sufficient? It might obscure some unintended behavior though. Either way a comment about this check might be helpful.

Reply:
I think you are right. That check was superflous and `strconv.Atoi()` will normalize it anyway.